mender-auth/main.cpp

```diff
@@ -0,0 +1,26 @@
+#include <iostream>
+
+using namespace std;
+
+
+#include <common/json/json.hpp>
+
+// This method only knows about the interface Json
+void hello_world(json::Json *j)
```

Comment:
I think this is unused, or? (same for the other one)

In any case, I was going to comment on the raw pointer, which should be replaced, most likely by `shared_ptr`.

Reply:
Thank you, it was meant to be used by the main below! Otherwise the whole point of this demo is lost :see_no_evil: 

The idea is to showcase that this function only knows about the interface (pure abstract class) while the caller instantiates a concrete class.

I'll fix and also add the `shared_ptr` :+1: